### DISCUSSION TITLE
Component Migration to Version 4

### DISCUSSION THREAD

**Discussion by jschmi102** [joomla/joomla-cms#35657] [General]
hi,
I'm on my way to migrate a component from V3 to V4. And I had some problems for which I sofar did not find a solution - mainly because there is **no documentation at all** for these tasks. For examle I got stuck now with this:
V3 had a couple of very  helpful utility classes in behaviour.php - e.g. you could use formvalidation using "HTMLHelper::_('behavior.formvalidation');" - all this is gone, - where? I could not find out. Is there any doc for developers  to find a replacement or solution for replaced or changed APIs?  The only thing I found out sofar is that functions are gone - no doc for a replacement or a solution. This is really frustrating.

**Comment by brianteeman** [comment]
I would start out here https://docs.joomla.org/JDOC:Joomla_4_Tutorials_Project and pay specific attention to https://docs.joomla.org/Joomla_4_Tips_and_Tricks:_Form_Validation_Basics
